### ISSUE TITLE
Weird S_REGION for Spec2Pipeline resample_spec for 01539003001

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#7265]
_Issue [JP-2802](https://jira.stsci.edu/browse/JP-2802) was created on JIRA by [Brian Hayden](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bhayden):_

A help desk ticket came in, and [David Rodriguez](https://jira.stsci.edu/secure/ViewProfile.jspa?name=drodriguez) noticed some strange footprints for, specifically, jw01539003001_03102_00005_nrs2 and jw01539003001_03102_00001_nrs1. They seemed to span from one celestial pole to the other (-89 to 89).

I tracked down that jw01539003001_03102_00001_nrs1 is a member of jw01539-o003_20220720t061905_spec2_005, and checked the SDP log (attached), noticing the offending S_REGION occurs in the stpipe.Spec2Pipeline.resample_spec section.

 
```
2022-07-20 06:41:10,802 - stpipe.Spec2Pipeline.resample_spec - INFO - Update S_REGION to POLYGON ICRS  81.180590749 -89.374247193 81.112666873 -89.374247193 81.112666873 89.303900795 81.180590749 89.303900795
```

**Comment by stscijgbot-jp** [comment]
_Comment by [Richard A. Shaw](https://jira.stsci.edu/secure/ViewProfile.jspa?name=shaw) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=600562#comment-600562):_

Does it only affect Spec2 pipeline products? Or is the problem more wide-ranging? [David Rodriguez](https://jira.stsci.edu/secure/ViewProfile.jspa?name=drodriguez) performed a quick check of CAOM and this specific program seems to be the only one so far with these malformed footprints. But it would be nice to find the smoking gun, so to speak.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=606238#comment-606238):_

[Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva) Hoping someone in team periwinkle can look at this?

**Comment by stscijgbot-jp** [comment]
_Comment by [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=609424#comment-609424):_

It would help to have the data.

**Comment by stscijgbot-jp** [comment]
_Comment by [Nadia Dencheva](https://jira.stsci.edu/secure/ViewProfile.jspa?name=dencheva) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=610040#comment-610040):_

Note to developers: I think S_REGION is computed correctly in spec2 and spec3. It looks like the issue is with determining the output WCS of the resampled image. The one thing that is different from other Nirspec cases is that the data spans NRS1 and NRS2. I think the Spec2 products are correct, the issues is specifically with resample_spec for Nirspec when the observations are on both Nirspec SCAs.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=612642#comment-612642):_

Reprocessed the rate files through calwebb_spec2 using the latest Cal 1.8.2 (B9.0) and still getting bad S_REGION values in the s2d products.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=612739#comment-612739):_

Doing some analysis on a single s2d file created from a single rate file (running through the calwebb_spec2 pipeline) does show odd behavior of the WCS for the s2d file. Using the detector->world transform from the WCS object to simply compute ra, dec, lambda tuples for each pixel in the s2d data array, the RA/Dec values are sensible (i.e. what you would expect given the target RA/Dec) over most of the resampled array, but then suddenly start diverging wildly near the long-wavelength end. Thus when the footprint calculation simply finds the min/max RA/Dec values for the corners, it ends up with values that cover a huge patch of the sky. Must be something going wrong in the creation of the output wcs grid for the resampled array, within the resample_spec step.  Top people are investigating.

**Comment by stscijgbot-jp** [comment]
_Comment by [Mihai Cara](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mcara) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=617248#comment-617248):_

A fix is proposed in #7359

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2802?focusedCommentId=619167#comment-619167):_

Fixed by #7359
